src/sys/socket/sockopt.rs

```diff
@@ -478,6 +492,53 @@ unsafe impl<'a> Set<'a, usize> for SetUsize {
     }
 }
 
+/// Getter for a `OsString` value.
+struct GetOsString<T: AsRef<[u8]>> {
+    len: socklen_t,
+    val: T,
+}
+
+unsafe impl<T: AsRef<[u8]>> Get<OsString> for GetOsString<T> {
+    unsafe fn blank() -> Self {
+        GetOsString {
+            len: mem::size_of::<T>() as socklen_t,
+            val: mem::zeroed(),
+        }
+    }
+
+    fn ffi_ptr(&mut self) -> *mut c_void {
+        &mut self.val as *mut T as *mut c_void
+    }
+
+    fn ffi_len(&mut self) -> *mut socklen_t {
+        &mut self.len
+    }
+
+    unsafe fn unwrap(self) -> OsString {
+        OsStr::from_bytes(self.val.as_ref()).to_owned()
+    }
+}
+
+/// Setter for a `OsString` value.
+struct SetOsString<'a> {
+    val: &'a OsString,
```

Comment:
You should basically never use a reference to an `OsString`.  Instead, use its borrowed form: `&OsStr`.

Reply:
Done